code/edgebridge/src/test/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtensionTests.java

```diff
@@ -172,15 +518,128 @@ public void test_handleTrackRequest_dispatchesEdgeRequestEvent() {
 					"data",
 					new HashMap<String, Object>() {
 						{
-							put("action", "Test Action");
 							put(
-								"contextdata",
+								"__adobe",
 								new HashMap<String, Object>() {
 									{
-										put("testKey", "testValue");
+										put(
+											"analytics",
+											new HashMap<String, Object>() {
+												{
+													put("linkName", "action name");
+													put("linkType", "other");
+													put("events", "event1,event2,event3,event4,event12,event13");
+													put(
+														"products",
+														";product1;1;5.99;event12=5.99;evar5=merchEvar5,;product2;2;10.99;event13=6;eVar6=mercheVar6"
+													);
+													put("c1", "propValue1");
+													put("cc", "USD");
+													put(
+														"contextData",
+														new HashMap<String, String>() {
+															{
+																put("key1", "value1");
+															}
+														}
+													);
+												}
+											}
+										);
 									}
 								}
 							);
+							put("key2", "value2");
+						}
+					}
+				);
+				put(
+					"xdm",
+					new HashMap<String, Object>() {
+						{
+							put("eventType", EdgeBridgeTestConstants.JsonValues.EVENT_TYPE);
+							put(
+								"timestamp",
+								TimeUtils.getISO8601UTCDateWithMilliseconds(new Date(event.getTimestamp()))
+							);
+						}
+					}
+				);
+			}
+		};
+
+		assertEquals(expectedData, responseEvent.getEventData());
+		assertEquals(event.getUniqueIdentifier(), responseEvent.getParentID());
+	}
+
+	@Test
+	public void test_handleTrackEvent_trackState_dispatchesEdgeRequestEvent() {
+		final Event event = new Event.Builder("Test Track Event", EventType.GENERIC_TRACK, EventSource.REQUEST_CONTENT)
+			.setEventData(
+				new HashMap<String, Object>() {
+					{
+						put("state", "state name");
+						put("key2", "value2");
+						put(
+							"contextdata",
+							new HashMap<String, String>() {
+								{
+									put("&&events", "event1,event2,event3,event4,event12,event13");
+									put("&&c1", "propValue1");
+									put("&&v1", "evarValue1");
+									put("key1", "value1");
+								}
```

Comment:
Current tests look good, but would love to see some additional ones for negative testing, null/empties, key name "&&", other special characters